### DISCUSSION TITLE
Java client

### DISCUSSION THREAD

**Discussion by uwegeercken** [kuzudb/kuzu#1322] [General]
Just curious to understand if a Java client is already on the roadmap.

Uwe

**Comment by ray6080** [comment]
Hi @uwegeercken, adding support for Java apis are definitely on our roadmap, though we are not sure yet if we can get there in the next release. We will update here once we are getting started on it!

**Reply by semihsalihoglu-uw** [reply]
to be a bit more upfront, it's what we have in mind after node.js api, which we are looking into right now.

**Comment by uwegeercken** [comment]
thanks. that sounds good. I will wait for the day to come....

**Comment by AubreyRoatsIY** [comment]
Of course, it's always important to prioritize development efforts based on user needs and feasibility, so there may be other factors that need to be considered as well.

Overall, it's great to see people asking questions and expressing interest in new features. Who knows, maybe a Java client will be added to the roadmap soon!

**Reply by semihsalihoglu-uw** [reply]
Yes, it is in fact in our immediate roadmap (right after node.js) but it will still take sometime to get there but hopefully not too much time.

**Comment by prrao87** [comment]
@uwegeercken We do have a Java client and its documentation is here:
https://docs.kuzudb.com/client-apis/java/
